pincer/objects/guild/guild.py

```diff
@@ -474,19 +477,328 @@ async def modify_member(
+            Keys:
+                - id : :class:`~pincer.utils.snowflake.Snowflake`
+                - position : Optional[:class:`int`]
+                - lock_permissions : Optional[:class:`bool`]
+                - parent_id : Optional[:class:`~pincer.utils.snowflake.Snowflake`]
+        """
+        await self._http.patch(
+            f"guilds/{self.id}/channels",
+            data=channel,
+            headers=remove_none({"X-Audit-Log-Reason":reason})
+        )
+
+    async def list_active_threads(self) -> Tuple[
+        Generator[Union[PublicThread, PrivateThread]], Generator[GuildMember]]:
+        """|coro|
+        Returns all active threads in the guild, including public and private threads.
+        
+        Returns
+        -------
+        Generator[Union[:class:`~pincer.objects.guild.channel.PublicThread`, :class:`~pincer.objects.guild.channel.PrivateThread`]], Generator[:class:`~pincer.objects.guild.member.GuildMember`]]
+            The new member object.
+        """
+        data = await self._http.get(f"guilds/{self.id}/threads/active")
+
+        threads = (
+            Channel.from_dict(construct_client_dict(self._client, data))
+            for channel in data["threads"]
+        )
+        members = (
+            GuildMember.from_dict(construct_client_dict(self._client, data))
+            for member in data["members"]
+        )
+
+        return threads, members
+
+    async def list_guild_members(self, limit: int = 1, after: int = 0):
+        """|coro|
+        Returns a list of guild member objects that are members of the guild.
+
+        Parameters
+        ----------
+        limit : int
+            max number of members to return (1-1000) |default| :data:`1`
+        after : int
+            the highest user id in the previous page |default| :data:`0`
+        """
+
+        members = await self._http.get(
+            f"guilds/{self.id}/members?{limit=}&{after=}"
+        )
+        
+        for member in members:
+            yield GuildMember.from_dict(construct_client_dict(self._client, data))
+        
+
+    async def search_guild_members(
+        self, 
+        query: str,
+        limit: Optional[int] = None
+    ) -> AsyncGenerator[GuildMember, None]:
+        """|coro|
+        Returns a list of guild member objects whose username or nickname starts with a provided string.
+
+        Parameters
+        ----------
+        query : str
+            Query string to match username(s) and nickname(s) against.
+        limit : Optional[int]
+            max number of members to return (1-1000) |default| :data:`1`
+        Returns
+        -------
+        AsyncGenerator[:class:`~pincer.objects.guild.member.GuildMember`, :data:`None`]
+            list of guild member objects
+        """
+
+        data = await self._http.get(
+            f"guilds/{id}/members/search?{query=!s}"
+            (f"&{limit}" if limit else "")
+        )
```

Comment:
Missing `+` sign, now you're tryna call the string :skull:
```suggestion
        data = await self._http.get(
            f"guilds/{id}/members/search?{query=!s}"
            + (f"&{limit}" if limit else "")
        )
```

Reply:
bruh. i just did what you told me to do. your fault 100% ~~/s obv.~~